### PR TITLE
Insectoids Rework: The spritening 

### PR DESCRIPTION
**Pull request by Buchse**
I had pomf remove my insectoid race from the whitelist a while back, and have since been working on a fairly large rework for them. The largest portion of this PR is a graphical overhaul, completely changing the look of the insectoids, as well as adding in a LOT of alternate sprites for clothing of almost every kind. Adding onto this, i've removed the severe brute damage weakness they originally had after seeing how big of a handicap it is to a race with blood and bones. The burn mod remains in place. 
newinsectallsides
This is the new bodystyle, everything is properly chopped and set up correctly for the icobase to use. 
I'll also dump a selection of screenshots showing off the sheer number of alts i've set up. 
insectcloth1
insectcloth2
insectear1
insectglasses
insecthats1
insecthats2
insecthats3
insecthats4
insectmasks
insectshoes
insectsuit1
insectsuit2

Insectoids will also have a mild accent similar to plasmamen, in which the letter "s" is replaced by the letter "z". 
insectoid speech replacement example

I fixed the bug making female insectoids completely invisible, which was a simple icobase error. 
There's more content planned for them, especially when it comes to alternate sprites and customization options. 

[sprites]
[tested]

:cl:
 * rscadd: Replaced insectoid body sprites, added a shitton of alternate clothing sprites and the code for them, added bug accents. (giant overhaul)
 * rscdel: Renamed old insectoid sprites to 'r_insectoidold' and 'r_def_insectoidold' respectively, removed insect ability to wear balaclavas because I couldn't figure out a way to sprite an alt and have it not look extremely stupid. 

* tweak: Insectoid brute modifier reduced from 130% to 100%.
* tweak: Insectoid burn modifier reduced from 130% to 110%

### CODE REVIEW

**Comment by SageOfLight** [comment]
I am here to scuttle.

**Comment by oversane** [comment]
stupid fuckin bug i want to crush em
i approve

**Comment by tuypo1** [comment]
Are they still completely rad Immune? I think there needs to be something done to make them at least slightly in danger from dragging shards around, maybe have rads not able to hurt them but able to slow them down, like an organ that allows fast movement but gets damaged by rads

**Comment by Buchse** [comment]
> Are they still completely rad Immune? I think there needs to be something done to make them at least slightly in danger from dragging shards around, maybe have rads not able to hurt them but able to slow them down, like an organ that allows fast movement but gets damaged by rads

They're rad resistant, but not overall immune. They ignore the initial rad damage, and have a bulwark against the toxins, but they'll get murked by IBs soon enough if they don't get help

**Comment by Myrios** [comment]
> > Are they still completely rad Immune? I think there needs to be something done to make them at least slightly in danger from dragging shards around, maybe have rads not able to hurt them but able to slow them down, like an organ that allows fast movement but gets damaged by rads
> 
> They're rad resistant, but not overall immune. They ignore the initial rad damage, and have a bulwark against the toxins, but they'll get murked by IBs soon enough if they don't get help

Add on top of that organ damage from radiation and all that entails so even chugging gallons of bicard wont save you from prolonged exposure.

**Comment by ShiftyRail** [comment]
Rather strange that they can use the hardsuits of humans without needing any adjustment.
I won't complain, though, this makes my job as coders to outfits much easier.

**Comment by PrimeDSS13** [comment]
can you still change eye color or is it a set black now?

**Comment by jknpj** [comment]
looks like a capybara

**Comment by Buchse** [comment]
> can you still change eye color or is it a set black now?

You can in fact still change eyecolor

**Comment by Gazathor** [comment]
I for one welcome our new bug janitors

**Comment by Waismuth** [comment]
I like the idea of a new playable race, thought they should be different in playstyle, because diona already have rad and toxin shielding. Maybe make em similar to flypeople? They can't ingest food normally so upon eating something they regurgitate it and suck up the barf, so the ability of feeding on vomit, or maybe give them low vision/change the way the see the world? One anon suggested them having 4 arms.

**Comment by Buchse** [comment]
> I like the idea of a new playable race, thought they should be different in playstyle, because diona already have rad and toxin shielding. Maybe make em similar to flypeople? They can't ingest food normally so upon eating something they regurgitate it and suck up the barf, so the ability of feeding on vomit, or maybe give them low vision/change the way the see the world? One anon suggested them having 4 arms.

I'm deliberately trying to keep them away from the "haha roach nasty trash" stuff tbh, it only leads to silly shitposting. 
In regards to them having 4 arms, while we do support mobs having multiple arms, it's janky in every other regard. We don't have support for how gloves work with multiple arms, and we don't have support for targeting limbs that aren't two arms and two legs with our targeting doll.

File: code/modules/mob/living/carbon/species.dm [hunk 1]
```diff
@@ -1293,13 +1293,25 @@ var/list/has_died_as_golem = list()
...
-	burn_mod = 1.3
+	burn_mod = 1.1
 	tox_mod = 0.5
 
 	blood_color = "#ebece6"
 	flesh_color = "#9c7f25"
 
+	uniform_icons = 'icons/mob/species/insectoid/uniform.dmi'
+//	fat_uniform_icons = 'icons/mob/uniform_fat.dmi'
+	gloves_icons    = 'icons/mob/species/vox/gloves.dmi'
```
Comment (Kurfursten) [inline code/modules/mob/living/carbon/species.dm:1304]: are you sure?

File: code/modules/mob/living/carbon/species.dm [hunk 2]
```diff
@@ -1293,13 +1293,25 @@ var/list/has_died_as_golem = list()
 	anatomy_flags = HAS_LIPS | HAS_SWEAT_GLANDS
 
 	default_mutations=list(RAD_IMMUNE)
-	brute_mod = 1.3
-	burn_mod = 1.3
+	burn_mod = 1.1
```
Comment (Kurfursten) [inline code/modules/mob/living/carbon/species.dm:1296]: lowering burn mod is undocumented balance change

File: code/modules/mob/living/carbon/species.dm [hunk 3]
```diff
@@ -1293,13 +1293,25 @@ var/list/has_died_as_golem = list()
 	anatomy_flags = HAS_LIPS | HAS_SWEAT_GLANDS
 
 	default_mutations=list(RAD_IMMUNE)
-	brute_mod = 1.3
```
Comment (Kurfursten) [inline code/modules/mob/living/carbon/species.dm:1296]: removing brute mod should receive its own line in changelog and note the degree of buff (insectoids now take 100% brute damage, down from 130%)

File: code/modules/mob/living/carbon/species.dm [hunk 4]
```diff
@@ -1310,10 +1322,13 @@ var/list/has_died_as_golem = list()
...
 		"eyes" =     /datum/organ/internal/eyes/compound/
 		)
+/datum/species/insectoid/handle_speech(var/datum/speech/speech, mob/living/carbon/human/H)
+	speech.message = replacetext(speech.message, "s", "z") //stolen from plasman code if it borks.
+	..()
 
 	species_intro = "You are an Insectoid.<br>\
-					Your body is utterly immune to the perils of radiation, and you are able to better defend against toxic chemicals <br>\
-					However, your rigid body is somewhat more fragile than that of more soft-bodied species. Resilient though you may be, a good smack may put you out of commission."
+					Your body is highly resistant to the initial effects of radiation exposure, and you are able to better defend against toxic chemicals <br>\
```
Comment (Kurfursten) [inline code/modules/mob/living/carbon/species.dm]: period

File: code/modules/mob/living/carbon/species.dm [hunk 5]
```diff
@@ -1310,10 +1322,13 @@ var/list/has_died_as_golem = list()
...
+					However, your body is somewhat more susceptible to heat than that of other species. Resilient though you may be, burns will prove more dangerous than typical. ."
```
Comment (Kurfursten) [inline code/modules/mob/living/carbon/species.dm]: stilted grammar and double period

File: code/modules/clothing/masks/miscellaneous.dm [hunk 6]
```diff
@@ -79,6 +79,7 @@
 	body_parts_covered = HIDEHAIR | MOUTH
 	w_class = W_CLASS_SMALL
 	species_fit = list(VOX_SHAPED, GREY_SHAPED)
+	species_restricted = list("exclude", INSECT_SHAPED)
```
Comment (Kurfursten) [inline code/modules/clothing/masks/miscellaneous.dm:82]: undocumented change? insectoids not allowed to wear balaclavas... why?

**Comment by oversane** [comment]
> Rather strange that they can use the hardsuits of humans without needing any adjustment.
> I won't complain, though, this makes my job as coders to outfits much easier.

Vox can put on hardsuits too, can't they?

**Comment by Buchse** [comment]
> Rather strange that they can use the hardsuits of humans without needing any adjustment.
> I won't complain, though, this makes my job as coders to outfits much easier.
> 
> Vox can put on hardsuits too, can't they?

nope, they're explicitly stopped from it

**Comment by cloutcode** [comment]
make the females bigger ahaha

**Comment by SkowronX** [comment]
fuckin send it

**Comment by Blithering** [comment]
They seem pretty vox shaped, how well do they fit in vox suits?

**Comment by yackemflam** [comment]
Remove the bottom part of the sec helmet, I will die on this hill that you can morph the shape but not change it to a completely different helmet

**Comment by Buchse** [comment]
> They seem pretty vox shaped, how well do they fit in vox suits?

they don't fit in them, vox suits have a tag restricting only vox to using them

**Comment by Buchse** [comment]
> Remove the bottom part of the sec helmet, I will die on this hill that you can morph the shape but not change it to a completely different helmet

no, be happy I was able to make it open-faced at all

**Comment by yackemflam** [comment]
No, telling someone to be happy with something isn't going to make them happy.
And two fuck you fix it or I will, why is sec the only department that get exceptions to this?

**Comment by Buchse** [comment]
> No, telling someone to be happy with something isn't going to make them happy.
> And two fuck you fix it or I will, why is sec the only department that get exceptions to this?

1: too bad
2: sec isn't the only department that get exceptions, literally every clothing item i've worked on so far changes to fit the bugs, sometimes drastically. The original sec helmet is annoying as hell to try and modify at all and still look good. This was a conscious design choice based on the fact that there wasn't a better way to do it.

**Comment by drawsstuff** [comment]
> No, telling someone to be happy with something isn't going to make them happy.
> And two fuck you fix it or I will, why is sec the only department that get exceptions to this?

Have you seen the others? Everything was changed to fit bug people, the hardsuit helmets look fucking wild man, in a good way.

**Comment by yackemflam** [comment]
the others doesn't change the entirety of the uniform, just changes the shape as if they're wearing the same clothes as us

**Pull request merged**